interfaces/builtin/multipass_support.go

```diff
@@ -0,0 +1,105 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const multipassSupportSummary = `allows operating as the Multipass service`
+
+const multipassSupportBaseDeclarationPlugs = `
+  multipass-support:
+    allow-installation: false
+    deny-auto-connection: true
+`
+
+const multipassSupportBaseDeclarationSlots = `
+  multipass-support:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const multipassSupportConnectedPlugAppArmor = `
+# Description: allow operating as the Multipass daemon. This policy intentionally
+# allows the Multipass daemon to configure AppArmor, and we expect Multipass to
+# create and apply correct policies for the child processes it spawns.
+
+# Allow socket
+/run/multipass_socket  rw,
+capability chown,
+capability fsetid,
+
+# Multipass generates AppArmor profiles for the child processes it spawns
+# Allow it to add these profiles and run the child under that profile.
+/usr/sbin/aa-exec ixr,
+/sbin/apparmor_parser ixr,
+/etc/apparmor.d/cache/ r,
+/etc/apparmor.d/cache/.features r,
+/etc/apparmor.d/{,cache/}multipass* rw,
+/etc/apparmor.d/tunables/{,**} r,
+/etc/apparmor.d/abstractions/{,**} r,
+/etc/apparmor/parser.conf r,
+/etc/apparmor/subdomain.conf r,
+/sys/kernel/security/apparmor/.replace w,
+/sys/kernel/security/apparmor/.remove w,
+/sys/kernel/security/apparmor/{,**} r,
+
+capability net_admin,
+capability mac_admin,
+capability kill, # multipass needs to be able to kill child processes
+
+# Profiles Multipass generates have a naming scheme. Need unsafe to prevent env scrubbing
+change_profile unsafe /var/snap/{@{SNAP_NAME},@{SNAP_INSTANCE_NAME}}/usr/bin/* -> multipass.*,
+signal (send) peer=multipass.*,
+ptrace (read, trace) peer=multipass.*,
+`
+
+const multipassSupportConnectedPlugSecComp = `
+# Description: allow operating as the Multipass daemon, and enable its various child
+# processes (qemu, qemu-img, dnsmasq)
+
+# Multipassd needs these. Qemu also could use them, but multipass will execute Qemu
+# with a custom profile that will lock down its filesystem access.
+accept
+accept4
+bind
+listen
+
+# dnsmasq fails unless it can drop supplementary groups
+setgroups 0 -
+setgroups32 0 -
```

Comment:
FYI, I don't know if you are patching dnsmasq for this, but calling setgroups() with zero groups (ie, `setgroups(0, NULL)`, which is what you want for the snap) is only guaranteed to work on Linux, so if you patched and that is meant to be portable, then you'll want to check if Linux and call in the above manner, otherwise when not Linux, call with size as '1' and give the egid as the first of the list of 1 supplementary groups.

Reply:
Good to know. We're just using dnsmasq from the repo in the snap. And we intend only to distribute multipass with the snap. On mac/windows we don't use dnsmasq at all, and don't plan unix support at this time.